cfme/v2v/migrations.py

```diff
@@ -0,0 +1,417 @@
+        """Add mapping from one or more sources to one target.
+
+        Args:
+            sources: list of source items
+            target: str containing target item
+        """
+
+        sources = sources or []
+        if isinstance(sources, list) and sources and target:
+            for item in sources:
+                self.logger.info("Mapping Source Cluster %s to Target Cluster %s", item, target)
+                self.source_clusters.select_by_text(item, partial_match=partial_match)
+                self.target_clusters.select_by_text(target, partial_match=partial_match)
+                self.add_mapping.click()
+
+
+class InfraMappingWizardDatastoresView(View):
+    include_buttons_set1 = View.include(InfraMappingCommonButtonsSet1)
+    include_buttons_set2 = View.include(InfraMappingCommonButtonsSet2)
+    source_datastores = MultiSelectList('source_datastores')
+    target_datastores = MultiSelectList('target_datastores')
+    cluster_selector = BootstrapSelect(locator='.//div[contains(@class,"cluster_select_select")]')
+
+    @property
+    def is_displayed(self):
+        return self.source_datastores.item_count > 0 or self.target_datastores.item_count > 0
+
+    def fill(self, values, partial_match=False):
+        """Use _add_cluster_mapping to add all mappings specified in values.
+
+        Args:
+        values:
+            format as following:
+                {
+                    'mapping1':
+                    {
+                        'sources':['item1','item2'],
+                        'target': 'item_target1'
+                    },
+                    'mapping2':
+                    {
+                        'sources':['item3','item4'],
+                        'target': 'item_target2'
+                    },
+                    ...
+                }
+        """
+        try:
+            for mapping in values:
+                self._add_datastore_mapping(values[mapping]['sources'],
+                    values[mapping]['target'], partial_match=partial_match)
+            return True
+        except Exception as e:
+            self.logger.debug("Following Exception occured while trying to fill:%s", e)
+            return False
+
+    def _add_datastore_mapping(self, sources=None, target=None, partial_match=False,
+            handle_alert=True):
+        """Add mapping from one or more sources to one target
+.
+        Args:
+            sources: list of source items
+            target: str containing target item
+        """
+        sources = sources or []
+        if isinstance(sources, list) and sources and target:
+            for item in sources:
+                self.source_datastores.select_by_text(item, partial_match)
+                self.target_datastores.select_by_text(target.split("(")[0],
+                    partial_match=partial_match)
+                if handle_alert and self.browser.alert_present:
+                    # TODO: Investigate why alert_present is False
```

Comment:
Because it's not a browser alert, but a flash message.

Reply:
How do we validate presence of flash?

Reply:
`widgetastic.patternfly` has `FlashMessage` class, but it doesn't work here, so you can create a special widget for flash messages in this wizard.

Reply:
@quarckster thanks, I will keep this as TODO for this PR, since it is a corner case and I will take care of not letting my test case hit this. Even if I did hit, it won't fail test case. If that's alright.